remill/Arch/X86/Semantics/MMX.h

```diff
@@ -73,6 +149,784 @@ DEF_SEM(PUNPCKHQDQ, D dst, S1 src1, S2 src2) {
+	for(size_t i = 0; i < vec_count; i++) {
+		auto src_entry = SExtractV16(src_vec, i);
+		auto mask = SShr(src_entry, decltype(src_entry)(15));
+		auto abs_value = SSub(SXor(src_entry, mask), mask);
+		dst_vec = SInsertV16(dst_vec, i, abs_value);
+	}
+	SWriteV16(dst, dst_vec);
+}
+DEF_ISEL(PABSW_MMXq_MMXq) = PABSW<V64W, V64>;
+DEF_ISEL(PABSW_MMXq_MEMq) = PABSW<V64W, MV64>;
+
+template <typename D, typename S1>
+DEF_SEM(PABSD, D dst, S1 src1) {
+	auto src_vec = SReadV32(src1);
+	auto dst_vec = SClearV32(SReadV32(dst));
+	auto vec_count = NumVectorElems(src_vec);
+	for(size_t i = 0; i < vec_count; i++) {
+		auto src_entry = SExtractV32(src_vec, i);
+		auto mask = SShr(src_entry, decltype(src_entry)(31));
+		auto abs_value = SSub(SXor(src_entry, mask), mask);
+		dst_vec = SInsertV32(dst_vec, i, abs_value);
+	}
+	SWriteV32(dst, dst_vec);
+}
+DEF_ISEL(PABSD_MMXq_MMXq) = PABSD<V64W, V64>;
+DEF_ISEL(PABSD_MMXq_MEMq) = PABSD<V64W, MV64>;
+
+// Need a better solution for handling PACKSS;
+// Soln : Sign extension and compare
+template <typename D, typename S1, typename S2>
+DEF_SEM(PACKSSWB, D dst, S1 src1, S2 src2) {
+    auto src1_vec = SReadV8(src1);
+	auto src2_vec = SReadV8(src2);
+	auto dst_vec = SClearV8(SReadV8(dst));
+
+	// Convert signed word to saturated signed byte
+	auto vec_count = NumVectorElems(src1_vec);
+	for(size_t i = 0; i < vec_count; i += 2) {
+		auto byte1 = SExtractV8(src1_vec, i);
+		auto byte2 = SExtractV8(src1_vec, i+1);
+		if (SCmpGt(byte2, decltype(byte2)(0))) {
+			byte1 = decltype(byte1)(0x7F);
+		}else if (SCmpEq(byte2, decltype(byte2)(0)) && SCmpLt(byte1, decltype(byte1)(0))) {
+			byte1 = decltype(byte1)(0x7F);
+		} else if (SCmpLt(byte2, decltype(byte2)(-1))){
+			byte1 = decltype(byte1)(0x80);
+		}
+		dst_vec = SInsertV8(dst_vec, i/2, byte1);
+	}
+	for(size_t i = 0; i < NumVectorElems(src2_vec); i += 2) {
+		auto byte1 = SExtractV8(src2_vec, i);
+		auto byte2 = SExtractV8(src2_vec, i+1);
+		if (SCmpGt(byte2, decltype(byte2)(0))) {
+			byte1 = decltype(byte1)(0x7F);
+		}else if (SCmpEq(byte2, decltype(byte2)(0)) && SCmpLt(byte1, decltype(byte1)(0))) {
+			byte1 = decltype(byte1)(0x7F);
+		} else if (SCmpLt(byte2, decltype(byte2)(-1))){
+			byte1 = decltype(byte1)(0x80);
+		}
+        dst_vec = SInsertV8(dst_vec, (i+vec_count)/2, byte1);
+	}
+	SWriteV8(dst, dst_vec);
+}
+DEF_ISEL(PACKSSWB_MMXq_MMXq) = PACKSSWB<V64W, V64, V64>;
+DEF_ISEL(PACKSSWB_MMXq_MEMq) = PACKSSWB<V64W, V64, MV64>;
+
+template <typename D, typename S1, typename S2>
+DEF_SEM(PACKSSDW, D dst, S1 src1, S2 src2) {
+    auto src1_vec = SReadV16(src1);
+	auto src2_vec = SReadV16(src2);
+	auto dst_vec = SClearV16(SReadV16(dst));
+    // Convert signed word to saturated signed byte
+    auto vec_count = NumVectorElems(src1_vec);
+	for(size_t i = 0; i < vec_count; i += 2) {
+		auto byte1 = SExtractV16(src1_vec, i);
+		auto byte2 = SExtractV16(src1_vec, i+1);
+		if (SCmpGt(byte2, decltype(byte2)(0))) {
+			byte1 = decltype(byte1)(0x7FFF);
```

Comment:
So, you can use something like `Maximize(val)` to get the maximum value for a type of `decltype(val)`.